### PR TITLE
feat: add support for `Python@3.12`

### LINKED ISSUE #1862: Enhancement: Python 3.12
### Summary

3.12.0b4 is underway and the release date for 3.12 quickly approaches.

Release Schedule: https://peps.python.org/pep-0693/
Available Betas/RCs: https://www.python.org/ftp/python/3.12.0/

Tooling is beginning to support this (https://github.com/astral-sh/ruff/commit/8bc7378002d080094d572e83f3a3a69df1023030), I imagine we will have a waiting game for some things.. but it would be nice to get a head start on this once 2.0 is released.

Support 3.12, adding it to our CI matrix version, testing, docs, etc.

## Pre-3.12 Release Pending Upstream Packages

- [x] `greenlet`
  - [x] https://github.com/python-greenlet/greenlet/pull/327
  - [x] Greenlet 3.0 release
- [x] `grpcio`
	- [x] https://github.com/grpc/grpc/issues/33063
- [x] `orjson >= 3.8.10`
    - [x] https://github.com/ijl/orjson/releases/tag/3.8.10

## Post-3.12 Release Pending Upstream Packages

- [x] https://github.com/MagicStack/uvloop/issues/567

 
--- 

<a href="https://polar.sh/litestar-org/litestar/issues/1862">
<picture>
  <source media="(prefers-color-scheme: dark)" srcset="https://polar.sh/api/github/litestar-org/litestar/issues/1862/pledge.svg?darkmode=1">
  Fund with Polar
</picture>
</a>

### PR DESCRIPTION
**Pull request by JacobCoffee**
### Pull Request Checklist

[//]: # "By submitting this issue, you agree to:"
[//]: # "- follow Litestar's [Code of Conduct](https://github.com/litestar-org/.github/blob/main/CODE_OF_CONDUCT.md)"
[//]: # "- follow Litestar's [Contribution Guidelines](https://litestar.dev/community/contribution-guide)"

- [ ] New code has 100% test coverage
- [ ] (If applicable) The prose documentation has been updated to reflect the changes introduced by this PR
- [ ] (If applicable) The reference documentation has been updated to reflect the changes introduced by this PR

### Description
[//]: # "Please describe your pull request for new release changelog purposes"
[//]: # "Example: 'This pr adds the capability to use server-sent events, and documents it in the usage docs'"

- Add Python 3.12 as a target - [3.12 Release Schedule](https://peps.python.org/pep-0693/#schedule) is slated to arrive within the next 3 months, with RC 1 coming at the first of next week

### Close Issue(s)
[//]: # "Please add in issue numbers this pull request will close, if applicable"
[//]: # "Examples: Fixes #4321 or Closes #1234"

- Closes #1862

### CODE REVIEW

**Review by sobolevn — APPROVED** [review]
Let's see if this works!

**Comment by JacobCoffee** [comment]
It did not 😓

**Comment by JacobCoffee** [comment]
image

File: .github/workflows/ci.yaml
Comment (provinzkraut) [inline .github/workflows/ci.yaml:1]: Need to adjust the coverage here to run on 3.12 instead of 3.11

**Review by provinzkraut — DISMISSED** [review]

**Comment by provinzkraut** [comment]
Seems like uvloop has some issues and no pre-built wheels?

**Comment by provinzkraut** [comment]
Seems to be blocked by https://github.com/MagicStack/uvloop/issues/567

File: .github/workflows/ci.yaml
Comment (JacobCoffee) [reply .github/workflows/ci.yaml:1]: @sobolevn mentioned bugs
https://github.com/litestar-org/litestar/pull/2063#discussion_r1343582777

File: tests/unit/test_logging/test_logging_config.py [hunk 1]
```diff
@@ -144,12 +145,18 @@ def test_root_logger(handlers: Any, listener: Any) -> None:
...
     "handlers, listener",
     [
-        [default_handlers, StandardQueueListenerHandler],
+        pytest.param(
+            default_handlers,
+            StandardQueueListenerHandler,
+            marks=pytest.mark.xfail(
+                condition=sys.version_info >= (3, 12), reason="change to QueueHandler/QueueListener config in 3.12"
+            ),
+        ),
```
Comment (peterschutt) [inline tests/unit/test_logging/test_logging_config.py:154]: Should circle back and resolve this - I'm not sure what this is testing TBH, but in 3.12 it raises a key error b/c it is expecting there to be a handler called `"cfg://handlers.console"` in `cfg://handlers`

File: pyproject.toml [hunk 2]
```diff
@@ -87,8 +88,8 @@ picologging = ["picologging"]
 prometheus = ["prometheus-client"]
 pydantic = ["pydantic[email]", "pydantic-extra-types"]
 redis = ["redis[hiredis]>=4.4.4"]
-sqlalchemy = ["advanced-alchemy>=0.2.2,<0.4.0"]
-standard = ["jinja2", "jsbeautifier", "uvicorn[standard]", "fast-query-parsers>=1.0.2"]
+sqlalchemy = ["advanced-alchemy==0.2.2"]
```
Comment (peterschutt) [inline pyproject.toml:91]:
```suggestion
sqlalchemy = ["advanced-alchemy>=0.2.2,<0.4.0"]
```

**Pull request merged**